Update server.py for Issue #25

arnabsinha99 opened the pull request:
Compared the strings in the respective files rather than reading the files every time the API was called.

theSage21 commented on server.py:
```diff
@@ -4,10 +4,9 @@
-
 app = Bottle()
 
+expected = open('./files/expected_outputs/expected_output.txt').read()
```
```suggestion
with open('./files/expected_outputs/expected_output.txt', 'r') as fl:
    expected = fl.read()
```

The other thing leaves the file open which is fine in this case but will lead to a bad habit later on.

theSage21 commented on server.py:
```diff
@@ -33,17 +32,9 @@ def download(number):
-    with open('./files/expected_outputs/expected_output.txt','r') as f2 , open(userpath,'r') as f1:
-        ans = True
-        ans = f1.read().strip() == f2.read().strip()
+    ans = (uploaded==expected)
```
since it's string comparison it might be a good idea to use `strip()` to avoid having mismatches due to extra `\n` in the end of any one file.

arnabsinha99 replied:
Yeah. I'll do it straight away.

arnabsinha99 replied:
Nope. It has to be done for uploaded as well. I have made the required changes.

theSage21 commented on server.py:
```diff
@@ -33,17 +33,10 @@ def download(number):
-        ans = True
-        ans = f1.read().strip() == f2.read().strip()
+    uploaded = request.files.get('upload').file.read() #uploaded outputs by user
+    expected.strip()
```
Python strings are immutable. Hence `x.strip()` does not change x at all. It produces a new string which is x after stripping. [Read more here](https://stackoverflow.com/questions/8680080/why-are-python-strings-immutable-best-practices-for-using-them)
```suggestion
    expected = expected.strip()
```

theSage21 commented on server.py:
```diff
@@ -33,17 +33,10 @@ def download(number):
+    uploaded.strip()
+    ans = (uploaded==expected)
 
     if ans==False:
```
Comparison with Booleans is not recommended.
```suggestion
    if not ans:
```

theSage21 commented:
Please mention what all issues you are solving in the Pull request description.

arnabsinha99 commented:
Changes made in this pull request:- 

1) Instead of loading the user output to the filesystem every time, I read the file and extracted the string globally so that disk space is not occupied unnecessarily. This string can now be used to compare with the expected output. 

2) The string from the expected output is also extracted so that we compare the strings itself in the API call rather than reading the file line by line.

3) The strings were stripped off white-spaces.

arnabsinha99 commented:
Changes made in pull request #33 :-

1)  Instead of loading the user output to the filesystem every time, I read the file and extracted the string globally so that disk space is not occupied unnecessarily. This string can now be used to compare with the expected output.

 2) The string from the expected output is also extracted so that we compare the strings itself in the API call rather than reading the file line by line.

3) The strings were stripped off white-spaces.

4) The expected output for the sample case was filled in the `expected_output.txt` file
